api/envoy/config/trace/v2/trace.proto

```diff
@@ -123,6 +123,12 @@ message OpenCensusConfig {
   // The Cloud project_id to use for Stackdriver tracing.
   string stackdriver_project_id = 4;
 
+  // (optional) By default, the Stackdriver exporter will connect to production
+  // Stackdriver. If stackdriver_address is non-empty, it will instead connect
+  // to this address, which is in the gRPC format:
+  // https://github.com/grpc/grpc/blob/master/doc/naming.md
+  string stackdriver_address = 10;
```

Comment:
How does this work? I'm a bit worried about potential blocking behavior on the data path (or just about anywhere in Envoy). Presumably this is using Google gRPC?

Reply:
It is using Google gRPC library directly. It's pulling the code out of the opencensus library to Envoy to be able to customize the channel creation. Tracing push/pull is not on the datapath, right?

Reply:
`grpc::CreateChannel` and `TraceService::NewStub` happen during the `Driver()` constructor but neither should be waiting on the network. Exporting spans to the service happens in a background thread as discussed before (and that's when and where we'll find out if e.g. the dest port is closed)

Reply:
OK, thanks for the explanation. There are a bunch of curl/gRPC things here that probably could be more optimally integrated with Envoy, but this is probably not a new issue in this trace extension.